### ISSUE TITLE
Build broken on Linux x86 + musl

### ISSUE THREAD

**Issue by andreastt** [mozilla/geckodriver#495]
https://travis-ci.org/mozilla/geckodriver/jobs/205053847

**Comment by andreastt** [comment]
Unblocking this for the 0.15.0 release.

**Comment by pbi-qfs** [comment]
Is this also the reason for #533 ?

**Comment by andreastt** [comment]
Yes.

**Comment by andreastt** [comment]
I must have had cached objects previously when I test-compiled i686-unknown-linux-musl on my workstation, as I cannot seem to make it work on a clean build; neither for debug or release.

What we thought was a Travis-specific issue appears to reproduce on Rust stable:

```
% rustc --version
rustc 1.15.1 (021bd294c 2017-02-08)
% cargo clean
% RUST_BACKTRACE=1 cargo build --target i686-unknown-linux-musl
   Compiling semver v0.1.20
   Compiling cfg-if v0.1.0
   Compiling libc v0.2.21
   Compiling gcc v0.3.43
   Compiling regex-syntax v0.4.0
   Compiling num_cpus v1.3.0
   Compiling term_size v0.2.3
   Compiling utf8-ranges v1.0.0
   Compiling unicode-normalization v0.1.4
   Compiling bzip2-sys v0.1.5
   Compiling rustc-demangle v0.1.4
   Compiling slog v1.5.2
   Compiling rustc_version v0.1.7
   Compiling crossbeam v0.2.10
   Compiling unicode-width v0.1.4
   Compiling unicase v1.4.0
   Compiling lazy_static v0.1.16
   Compiling hyper v0.10.5
   Compiling void v1.0.2
   Compiling unreachable v0.1.1
   Compiling winapi-build v0.1.1
   Compiling traitobject v0.1.0
   Compiling slog-atomic v0.4.3
   Compiling log v0.3.7
   Compiling num-traits v0.1.37
   Compiling rustc-serialize v0.3.22
   Compiling kernel32-sys v0.2.2
   Compiling backtrace v0.3.0
   Compiling rand v0.3.15
   Compiling unicode-segmentation v1.1.0
   Compiling num-integer v0.1.33
   Compiling backtrace-sys v0.1.10
   Compiling miniz-sys v0.1.9
   Compiling podio v0.1.5
   Compiling matches v0.1.4
   Compiling time v0.1.36
   Compiling unicode-bidi v0.2.5
   Compiling num-iter v0.1.33
   Compiling isatty v0.1.3
Build failed, waiting for other jobs to finish...
error: failed to run custom build command for `backtrace-sys v0.1.10`
process didn't exit successfully: `/home/ato/src/github.com/mozilla/geckodriver/target/debug/build/backtrace-sys-e7a3c65f1243f5fa/build-script-build` (exit code: 101)
--- stdout
OPT_LEVEL = Some("0")
TARGET = Some("i686-unknown-linux-musl")
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("i686-unknown-linux-musl")
TARGET = Some("i686-unknown-linux-musl")
HOST = Some("x86_64-unknown-linux-gnu")
CC_i686-unknown-linux-musl = None
CC_i686_unknown_linux_musl = None
TARGET_CC = None
CC = None
HOST = Some("x86_64-unknown-linux-gnu")
CROSS_COMPILE = None
TARGET = Some("i686-unknown-linux-musl")
HOST = Some("x86_64-unknown-linux-gnu")
CFLAGS_i686-unknown-linux-musl = None
CFLAGS_i686_unknown_linux_musl = None
TARGET_CFLAGS = None
CFLAGS = None
PROFILE = Some("debug")
running: "/home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-sys-0.1.10/src/libbacktrace/configure" "--with-pic" "--disable-multilib" "--disable-shared" "--disable-host-shared" "--target=i686-unknown-linux-musl" "--host=x86_64-unknown-linux-gnu"
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... i686-unknown-linux-musl
checking for x86_64-unknown-linux-gnu-gcc... musl-gcc
checking for C compiler default output file name... 

--- stderr
configure: WARNING: If you wanted to set the --build type, don't use --host.
    If a cross compiler is detected then cross compile mode will be used.
configure: error: in `/home/ato/src/github.com/mozilla/geckodriver/target/i686-unknown-linux-musl/debug/build/backtrace-sys-e4956d294e94bde6/out':
configure: error: C compiler cannot create executables
See `config.log' for more details.
thread 'main' panicked at 'failed with: exit code: 77', /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-sys-0.1.10/build.rs:134
stack backtrace:
   1:     0x564e71a1f50a - std::sys::imp::backtrace::tracing::imp::write::h3188f035833a2635
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2:     0x564e71a2589f - std::panicking::default_hook::{{closure}}::h6385b6959a2dd25b
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:349
   3:     0x564e71a2549e - std::panicking::default_hook::he4f3b61755d7fa95
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:365
   4:     0x564e71a25c87 - std::panicking::rust_panic_with_hook::hf00b8130f73095ec
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:553
   5:     0x564e71a25ac4 - std::panicking::begin_panic::h6227f62cb2cdaeb4
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:515
   6:     0x564e71a25a39 - std::panicking::begin_panic_fmt::h173eadd80ae64bec
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:499
   7:     0x564e719f56e4 - build_script_build::run::hfe2696f0d6fd0a85
                        at /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-sys-0.1.10/build.rs:134
   8:     0x564e719f3a0c - build_script_build::main::h8a6e4eb77bc1887a
                        at /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-sys-0.1.10/build.rs:74
   9:     0x564e71a2e85a - __rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libpanic_unwind/lib.rs:98
  10:     0x564e71a263c6 - std::rt::lang_start::h65647f6e36cffdae
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panicking.rs:434
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/panic.rs:351
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libstd/rt.rs:57
  11:     0x564e719f57a2 - main
  12:     0x7fdcd3cf72b0 - __libc_start_main
  13:     0x564e719e4319 - _start
  14:                0x0 - <unknown>
```

**Comment by andreastt** [comment]
Rust beta and stable fail with similar error messages, which means moving the target toolchain would not help us in this situation.

**Comment by jgraham** [comment]
Check for a rustc bug?

**Comment by andreastt** [comment]
For posterity, the relevant Debian packages are musl, musl-dev, and musl-tools.

**Comment by pbi-qfs** [comment]
Since this is officially fixed in 0.16 - can we expect to see a linux32 build of geckodriver 0.16 on the releases page?

**Comment by andreastt** [comment]
The 32-bit Linux CI build was fixed, but it turns out that it’s still failing—for other reasons—to compile in optimised mode. See https://github.com/mozilla/geckodriver/issues/666.
